### PR TITLE
fix(exec): apply filter rules to interleaved output in run_exec_impl

### LINKED ISSUE #1016: fix(exec): filter not applied to interleaved output, bypassing all filter rules
## Problem

All `exec_command` filter rules are silently bypassed for any command that writes to both stdout and stderr.

### Root cause

In `run_exec_impl` (`crates/aptu-coder/src/lib.rs`), lines are collected into `stdout_str`, `stderr_str`, and `interleaved_str` simultaneously. `apply_filter` runs afterward and updates only `output.stdout`. The response handler at the `exec_command` call site prefers `output.interleaved` when non-empty:

```rust
let output_text = if output.interleaved.is_empty() {
    format!("Stdout:\n{}\n\nStderr:\n{}", output.stdout, output.stderr)
} else {
    format!("Output:\n{}", output.interleaved)
};
```

Since `git pull` (and most git commands) write fetch progress to stderr and diff output to stdout, `interleaved` is never empty. The LLM always receives the unfiltered interleaved stream. The filtered `output.stdout` is never read.

This affects every filter rule (`git pull`, `git fetch`, `git push`, `git log`, `git status`, `git show`, `git commit`, `git diff`, `git add`, `cargo build`, `cargo test`).

### Observed symptom

`git pull` continues to consume large token counts despite PR #1014 adding filter rules and `--no-stat` injection. The `--no-stat` injection (`maybe_inject_no_stat`) does fire correctly and suppresses the diffstat at the git level. But the filter stripping of remaining output lines is never applied because the interleaved path bypasses it.

## Fix

Apply the filter to `interleaved_str` before it is stored in `output.interleaved`, using the same rule-matching logic already applied to `output.stdout`. The filter must run on the interleaved content, not only on the separated stdout.

Alternatively, rebuild `interleaved` from the already-filtered stdout and unfiltered stderr after `apply_filter` runs.

## Scope

- `crates/aptu-coder/src/lib.rs` -- filter application site in `run_exec_impl`
- `crates/aptu-coder/src/filters.rs` -- no change required to filter rules themselves
- Tests: add a test asserting that filter rules fire when stdout and stderr are both non-empty (interleaved path)

Closes this issue once `git pull` output is demonstrably reduced in token count on a repo with many changed files.

### PR DESCRIPTION
**Pull request by clouatre**
## Problem

All `exec_command` filter rules were silently bypassed for any command that writes to both stdout and stderr -- which includes virtually every `git` command.

In `run_exec_impl`, `apply_filter` updated only `output.stdout`. The response handler at the call site prefers `output.interleaved` when non-empty:

```rust
let output_text = if output.interleaved.is_empty() {
    format!("Stdout:\n{}\n\nStderr:\n{}", output.stdout, output.stderr)
} else {
    format!("Output:\n{}", output.interleaved)
};
```

`output.interleaved` is never empty for commands that write fetch progress to stderr and diff output to stdout (e.g. `git pull`). The LLM therefore always received the unfiltered interleaved stream, rendering every filter rule -- including the `--no-stat` injection added in #1014 -- ineffective.

Root cause confirmed by inspection: `filter_applied` was set correctly, but the content the LLM consumed was sourced from `output.interleaved`, which was never passed through `apply_filter`.

## Fix

In the existing filter application block in `run_exec_impl`, after updating `output.stdout`, apply the same compiled rule to `output.interleaved`:

```rust
output.stdout = filtered_stdout;
// Also filter interleaved: the response handler prefers interleaved when
// non-empty (which it always is for commands that write to both streams),
// so filtering only stdout would leave the LLM-visible output unfiltered.
// apply_filter is called separately on each field; there is no double-filtering
// because stdout and interleaved are independent strings assembled from the
// same source lines -- updating one does not affect the other.
output.interleaved = apply_filter(compiled_rule, &output.interleaved);
```

Single-site change. Reuses the already-matched `compiled_rule`. No signature changes to `apply_filter`, no new struct fields on `ShellOutput`.

## Verification

Smoke-tested against the patched binary: `git pull` on an up-to-date repo returns `filter_applied: "git pull: strip diff-stat noise"` and a clean interleaved string with no diff-stat table lines. The unit tests below cover the stripping path with actual diff-stat content.

## Tests

- `test_filter_applied_to_interleaved_with_both_streams`: strip pattern removes `| 42 +++` lines from interleaved while preserving stderr-origin lines (`From https://...`).
- `test_on_empty_substitution_in_interleaved`: when the filter strips all lines from interleaved, the `on_empty` substitution text is returned rather than an empty string.

Closes #1016

### CODE REVIEW

File: crates/aptu-coder/src/lib.rs [hunk 1]
```diff
@@ -3663,6 +3663,7 @@ async fn run_exec_impl(
             if compiled_rule.pattern.is_match(&command) {
                 let filtered_stdout = apply_filter(compiled_rule, &output.stdout);
                 output.stdout = filtered_stdout;
+                output.interleaved = apply_filter(compiled_rule, &output.interleaved);
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/src/lib.rs:3672]: Addressed: added a comment at the call site explaining that stdout and interleaved are independent strings assembled from the same source lines, so calling apply_filter on each is not double-filtering -- updating one does not affect the other.

File: crates/aptu-coder/src/lib.rs [hunk 2]
```diff
@@ -3663,6 +3663,13 @@ async fn run_exec_impl(
             if compiled_rule.pattern.is_match(&command) {
                 let filtered_stdout = apply_filter(compiled_rule, &output.stdout);
                 output.stdout = filtered_stdout;
+                // Also filter interleaved: the response handler prefers interleaved when
+                // non-empty (which it always is for commands that write to both streams),
+                // so filtering only stdout would leave the LLM-visible output unfiltered.
+                // apply_filter is called separately on each field; there is no double-filtering
+                // because stdout and interleaved are independent strings assembled from the
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/src/lib.rs:3670]: Confirmed. The existing code comment at the call site now documents this explicitly: stdout and interleaved are independent strings built from the same source lines, so apply_filter on each is not double-filtering.
